### PR TITLE
Refactor IsOwner/IsMember and use AccessListMember object.

### PR DESCRIPTION
**Pull request by mdwn**
IsOwner/IsMember have been renamed to IsAccessListOwner and IsAccessListMember for better clarity, and IsAccessListMember uses the new AccessListMember object now. References to IsMember in the OSS codebase have been replaced with IsAccessListMember, and tests have been updated.

### CODE REVIEW

File: lib/services/access_list.go [hunk 1]
```diff
@@ -206,6 +213,34 @@ func IsMember(identity tlsca.Identity, clock clockwork.Clock, accessList *access
...
+	member, err := memberGetter.GetAccessListMember(ctx, accessList.GetName(), username)
+	if trace.IsNotFound(err) {
+		// The member has not been found, so we know they're not a member of this list.
+		return trace.NotFound("user %s is not a member of the access list", username)
+	} else if err != nil {
+		// Some other error has occurred
+		return trace.Wrap(err)
+	}
+
+	expires := member.Spec.Expires
```
Comment (tigrato) [inline lib/services/access_list.go:229]: Is it possible to have expiration `member.Spec.Expires` after the next audit date? 
If yes, should we consider the expiration as the next audit date?

Comment (mdwn) [reply lib/services/access_list.go:229]: I think that's a good idea. I'll do a minimum of expires vs. spec expiration here.

File: lib/services/access_list.go [hunk 2]
```diff
@@ -206,6 +213,34 @@ func IsMember(identity tlsca.Identity, clock clockwork.Clock, accessList *access
...
+	if expires.IsZero() || accessList.Spec.Audit.NextAuditDate.Before(expires) {
+		expires = accessList.Spec.Audit.NextAuditDate
+	}
+
+	if !clock.Now().Before(expires) {
+		return trace.NotFound("user %s's membership has expired in the access list", username)
```
Comment (tigrato) [inline lib/services/access_list.go:235]: Should this be considered a not found error?
Once the membership expires should we use a `AccessDenied` instead?

File: lib/services/access_list.go [hunk 3]
```diff
@@ -166,7 +166,13 @@ func UnmarshalAccessListMember(data []byte, opts ...MarshalOption) (*accesslist.
 }
 
 // IsOwner will return true if the user is an owner for the current list.
+// TODO(mdwn): Remove this.
```
Comment (capnspacehook) [inline lib/services/access_list.go:169]: Is this being kept for now so `e` builds don't fail?

**Review by capnspacehook — APPROVED** [review]

File: lib/services/access_list.go (hunk 3 above)
Comment (jakule) [reply lib/services/access_list.go:169]: Good point. Can you add more context? Why this cannot be removed now? Should be removed on the next release...tomorrow?

File: lib/auth/userloginstate/generator_test.go [hunk 4]
```diff
@@ -258,13 +259,32 @@ func newAccessList(t *testing.T, name string, members []string, roles []string,
...
+func newAccessListMembers(t *testing.T, accessList string, members ...string) []*accesslist.AccessListMember {
+	alMembers := make([]*accesslist.AccessListMember, len(members))
+	for i, member := range members {
+		var err error
+		alMembers[i], err = accesslist.NewAccessListMember(header.Metadata{
+			Name: member,
+		}, accesslist.AccessListMemberSpec{
+			AccessList: accessList,
+			Name:       member,
+			Joined:     time.Now(),
```
Comment (jakule) [inline lib/auth/userloginstate/generator_test.go:277]: Shouldn't a fake clock be used here?

File: lib/auth/userloginstate/generator_test.go [hunk 5]
```diff
@@ -219,26 +230,16 @@ func initGeneratorSvc(t *testing.T) (*Generator, *svc) {
...
-	}
-
 	accessList, err := accesslist.NewAccessList(header.Metadata{
 		Name: name,
 	}, accesslist.Spec{
 		Title: "title",
 		Audit: accesslist.Audit{
-			Frequency: time.Hour,
+			Frequency:     time.Hour,
+			NextAuditDate: time.Now().Add(time.Hour * 48),
```
Comment (jakule) [inline lib/auth/userloginstate/generator_test.go:242]: This is confusing. If the review frequency is 1 hour why the next audit should be in 48h? :thinking:

File: lib/services/access_list.go [hunk 6]
```diff
@@ -206,6 +213,34 @@ func IsMember(identity tlsca.Identity, clock clockwork.Clock, accessList *access
...
+	if expires.IsZero() || accessList.Spec.Audit.NextAuditDate.Before(expires) {
+		expires = accessList.Spec.Audit.NextAuditDate
+	}
+
+	if !clock.Now().Before(expires) {
+		return trace.AccessDenied("user %s's membership has expired in the access list", username)
+	}
+
+	if !userMeetsRequirements(identity, accessList.Spec.MembershipRequires) {
+		return trace.AccessDenied("user %s does not meet membership requirements", username)
```
Comment (jakule) [inline lib/services/access_list.go:239]: I expect that this value is returned to a user. "does not meet membership requirements" is not very "self-explanatory". I think that the error message should be clearer.

File: lib/services/access_list_test.go [hunk 7]
```diff
@@ -392,6 +393,157 @@ func TestIsMember(t *testing.T) {
...
+			identity: tlsca.Identity{
+				Username: member3,
+				Groups:   []string{"mrole1", "mrole2"},
+				Traits: map[string][]string{
+					"mtrait1": {"mvalue1", "mvalue2"},
+					"mtrait2": {"mvalue3", "mvalue4"},
+				},
+			},
+			currentTime: time.Date(2023, 2, 1, 0, 0, 0, 0, time.UTC),
+			errAssertionFunc: func(t require.TestingT, err error, i ...interface{}) {
```
Comment (jakule) [inline lib/services/access_list_test.go:456]: nit:
```suggestion
			errAssertionFunc: func(t require.TestingT, err error, i ...any) {
```

**Review by jakule — APPROVED** [review]

File: lib/auth/userloginstate/generator_test.go (hunk 4 above)
Comment (mdwn) [reply lib/auth/userloginstate/generator_test.go:277]: The way these tests are designed it doesn't really matter, but may as well.

File: lib/services/access_list.go (hunk 3 above)
Comment (mdwn) [reply lib/services/access_list.go:169]: Yes it is. I'll add more context here.

**Review by tigrato — APPROVED** [review]

**Pull request merged**